### DISCUSSION TITLE
Should async: false force return a type "string" instead of "string | Promise<string>"?

### DISCUSSION THREAD

**Discussion by Jakobud** [markedjs/marked#3504] [General]
I am using marked with default options. Async is false by default. When using it in TypeScript I get

`Type 'string | Promise<string>' is not assignable to type 'string'`

![image](https://github.com/user-attachments/assets/ee39b269-3822-487b-84eb-aba0fb19cb92)

It seems that the type signature should return `string` when `async: false` and `Promise<string>` when `async: true`.

Am I misunderstanding something about this?

**Comment by UziTech** [comment]
Yes, you are missing that typescript cannot know if an extension changes async to true

```js
marked.use({async: true});

marked.parse(...) // returns Promise<string>
```

**Reply by Jakobud** [reply]
Oh yeah I guess that makes sense because the change happens at runtime. Doh! Thanks
